lib/services/local/sessiontracker.go

```diff
@@ -116,16 +117,46 @@ func (s *sessionTracker) GetActiveSessionTrackers(ctx context.Context) ([]types.
 		return nil, trace.Wrap(err)
 	}
 
-	sessions := make([]types.SessionTracker, len(result.Items))
-	for i, item := range result.Items {
+	sessions := make([]types.SessionTracker, 0)
+	expired := make([]backend.Item, 0)
+	now := time.Now().UTC()
+	for _, item := range result.Items {
 		session, err := unmarshalSession(item.Value)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
 
-		sessions[i] = session
+		switch {
+		case session.Expiry().After(now):
+			// Keep any items that aren't expired.
+			sessions = append(sessions, session)
+		case item.Expires.IsZero():
+			// Clear item if expiry is not set.
+			// We shouldn't need this ideally but we currently do not set
+			// tracker expiries. We will however do this in the future.
+			expired = append(expired, item)
+		default:
+			// If the tracker has expired and there is an expiry set, we can never take this branch
+			// as the backend implementation will not return the key anyway.
```

Comment:
Perhaps "the backend should not have returned this key, but the backend is responsible for cleaning up its own expired items"?